contracts/test/integration/IntegrationTestVoltMigratorRouter.t.sol

```diff
@@ -0,0 +1,162 @@
+        newVolt = new VoltV2(MainnetAddresses.CORE);
+
+        // Deploy new PSMs
+        PegStabilityModule.OracleParams memory oracleParamsUsdc;
+        PegStabilityModule.OracleParams memory oracleParamsDai;
+
+        oracleParamsUsdc = PegStabilityModule.OracleParams({
+            coreAddress: address(core),
+            oracleAddress: address(oracle),
+            backupOracle: address(0),
+            decimalsNormalizer: 12,
+            doInvert: true,
+            volt: IVolt(address(newVolt))
+        });
+
+        oracleParamsDai = PegStabilityModule.OracleParams({
+            coreAddress: address(core),
+            oracleAddress: address(oracle),
+            backupOracle: address(0),
+            decimalsNormalizer: 0,
+            doInvert: true,
+            volt: IVolt(address(newVolt))
+        });
+
+        usdcPSM = new PriceBoundPSM(
+            voltUsdcFloorPrice,
+            voltUsdcCeilingPrice,
+            oracleParamsUsdc,
+            0,
+            0,
+            reservesThreshold,
+            10_000e18,
+            10_000_000e18,
+            IERC20(address(usdc)),
+            compoundUsdcPcvDeposit
+        );
+
+        daiPSM = new PriceBoundPSM(
+            voltDaiFloorPrice,
+            voltDaiCeilingPrice,
+            oracleParamsDai,
+            0,
+            0,
+            reservesThreshold,
+            10_000e18,
+            10_000_000e18,
+            IERC20(address(dai)),
+            compoundUsdcPcvDeposit
+        );
+
+        // Deploy TempCoreRef
+        coreRef = new TempCoreRef(
+            MainnetAddresses.CORE,
+            IVolt(address(newVolt))
+        );
+
+        // Deploy Volt Migrator
+        voltMigrator = new VoltMigrator(
+            MainnetAddresses.CORE,
+            address(newVolt)
+        );
+
+        // Deploy Migrator Router
+        migratorRouter = new MigratorRouter(
+            address(newVolt),
+            address(voltMigrator),
+            daiPSM,
+            usdcPSM
+        );
+
+        // Grant stablecoin balances to PSMs
+        uint256 balance = usdc.balanceOf(MainnetAddresses.MAKER_USDC_PSM);
+        vm.prank(MainnetAddresses.MAKER_USDC_PSM);
+        usdc.transfer(address(usdcPSM), balance);
+
+        balance = dai.balanceOf(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        vm.prank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(daiPSM), balance);
+
+        // Grant old volt balance to user, and new volt balance to migrator
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+        // mint old volt to user
+        oldVolt.mint(address(this), mintAmount);
+        // mint new volt to migrator contract
+        newVolt.mint(address(voltMigrator), mintAmount);
+        vm.stopPrank();
+
+        // approve migratorRouter to use users old volt
+        oldVolt.approve(address(migratorRouter), type(uint256).max);
+    }
+
```

Comment:
let's add assertions here to ensure that PSM's and allocator are properly set up here.